newIDE/app/src/Profile/ChangeEmailDialog.js

```diff
@@ -22,72 +23,87 @@ type Props = {|
   error: ?AuthError,
 |};
 
-type State = {|
-  form: ChangeEmailForm,
-|};
+const ChangeEmailDialog = ({
+  onClose,
+  onChangeEmail,
+  firebaseUser,
+  changeEmailInProgress,
+  error,
+}: Props) => {
+  const [email, setEmail] = React.useState(firebaseUser.email);
+  const [isEmailValid, setIsEmailValid] = React.useState<boolean>(true);
 
-export default class ChangeEmailDialog extends Component<Props, State> {
-  state = {
-    form: {
-      email: this.props.firebaseUser.email,
-    },
-  };
+  const doChangeEmail = async () => {
+    const trimmedEmail = email.trim();
+    setEmail(trimmedEmail);
+    setIsEmailValid(emailRegex.test(trimmedEmail));
 
-  _onChangeEmail = () => {
-    if (this.props.changeEmailInProgress) return;
+    if (changeEmailInProgress || !email || !isEmailValid) return;
 
-    const { form } = this.state;
-    this.props.onChangeEmail(form);
+    await onChangeEmail({
+      email,
+    });
   };
 
-  render() {
-    const { onClose, changeEmailInProgress, error } = this.props;
-    const actions = [
-      <FlatButton
-        label={<Trans>Back</Trans>}
-        disabled={changeEmailInProgress}
-        key="back"
-        primary={false}
-        onClick={onClose}
-      />,
-      <LeftLoader isLoading={changeEmailInProgress} key="change-email">
-        <DialogPrimaryButton
-          label={<Trans>Save</Trans>}
-          primary
-          onClick={this._onChangeEmail}
+  return (
+    <Dialog
+      title={<Trans>Change your email</Trans>}
+      actions={[
+        <FlatButton
+          label={<Trans>Back</Trans>}
           disabled={changeEmailInProgress}
-        />
-      </LeftLoader>,
-    ];
-
-    return (
-      <Dialog
-        title={<Trans>Change your email</Trans>}
-        actions={actions}
-        maxWidth="sm"
-        cannotBeDismissed={changeEmailInProgress}
-        onRequestClose={onClose}
-        onApply={this._onChangeEmail}
-        open
+          key="back"
+          primary={false}
+          onClick={onClose}
+        />,
+        <LeftLoader isLoading={changeEmailInProgress} key="change-email">
+          <DialogPrimaryButton
+            label={<Trans>Save</Trans>}
+            primary
+            onClick={doChangeEmail}
+            disabled={changeEmailInProgress}
+          />
+        </LeftLoader>,
+      ]}
+      maxWidth="xs"
+      cannotBeDismissed={changeEmailInProgress}
+      onRequestClose={onClose}
+      onApply={doChangeEmail}
+      open
+    >
+      <form
+        onSubmit={event => {
+          event.preventDefault();
```

Comment:
maybe a little comment about why the preventDefault is needed could help in future debug sessions. (It is because it refreshes the page right?)
(and to add on all forms?)